File: csharp/sum-of-multiples/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        List<int> multipleNumbers = new List<int>();

        foreach (var multiple in multiples.Where(m => m > 0))
        {   
            var maxMultipleIndex = max / multiple - (max % multiple == 0 ? 1 : 0);

            multipleNumbers.AddRange(Enumerable.Range(1, maxMultipleIndex).Select(i => multiple * i));
        }

        return multipleNumbers.Select(x => x).Distinct().Sum();
    }
}

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit: int, multiples: list[int]) -> int:
    return sum({num for factor in multiples 
                if factor != 0 
                for num in range(factor, limit, factor)})

    # return sum({num for num in range(limit) 
    #     for factor in multiples 
    #     if factor != 0 and num % factor == 0})

    #if not multiples: return 0

    # multiples_tracked = set()

    # for num in multiples:
    #     if num != 0:
    #         multiples_tracked.update(range(num, limit, num))

    #return sum(multiples_tracked)  